PurchasesSwift/PurchasesSwift.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  PurchasesSwift.swift
+//  PurchasesSwift
+//
+//  Created by Andrés Boedo on 6/16/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@_exported import Purchases
+
+
+@objc public extension Purchases {
+    @objc func showManageSubscriptionModal() {
+
+        self.purchaserInfo { purchaserInfo, error in
+            if let error = error {
+                print("there was an error getting purchaserInfo: \(error.localizedDescription)")
+                return
+            }
+
+            guard let purchaserInfo = purchaserInfo else {
+                print("there was no error but purchaserInfo is null!")
+                return
+            }
+
+            guard let managementURL = purchaserInfo.managementURL else {
+                print("managementURL is nil, opening iOS subscription management page")
+                self.showAppleManageSubscriptions()
+                return
+            }
+
+            #if os(iOS)
+            if managementURL.isAppleSubscription() {
+                if #available(iOS 15.0, *) {
+                    detach {
+                        await self.showSK2ManageSubscriptions()
+                    }
+                }
+                return
+            }
+            #endif
+            self.openURL(managementURL)
+        }
+    }
+}
+
+public extension Purchases {
+
+    @available(iOS 9.0, *)
+    @available(macOS 10.12, *)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    func showAppleManageSubscriptions() {
+#if os(iOS)
+        if #available(iOS 15.0, *) {
+            detach {
+                await self.showSK2ManageSubscriptions()
+            }
+        } else {
+            self.openURL(.appleSubscriptionsURL)
+        }
+#elseif os(macOS)
+        self.openURL(.appleSubscriptionsURL)
+#endif
+    }
+
+    @MainActor
+    @available(iOS 15.0, *)
```

Comment:
would it potentially be possible to batch these into a single custom annotation? i can see us repeating these lines alllll over the place

Reply:
I _think_ we'd be able to do it at the extension declaration level, and it'd apply to everything in it.